plugin/pkg/shoot/seedmanager/admission.go

```diff
@@ -143,20 +154,53 @@ func (s *SeedManager) Admit(a admission.Attributes) error {
 }
 
 // determineSeed returns an appropriate Seed cluster (or nil).
-func determineSeed(shoot *garden.Shoot, lister gardenlisters.SeedLister) (*garden.Seed, error) {
-	list, err := lister.List(labels.Everything())
+func determineSeed(shoot *garden.Shoot, seedLister gardenlisters.SeedLister, shootLister gardenlisters.ShootLister) (*garden.Seed, error) {
+	seedList, err := seedLister.List(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
 
-	for _, seed := range list {
-		// We return the first matching seed cluster.
-		if seed.DeletionTimestamp == nil && seed.Spec.Cloud.Profile == shoot.Spec.Cloud.Profile && seed.Spec.Cloud.Region == shoot.Spec.Cloud.Region && seed.Spec.Visible != nil && *seed.Spec.Visible && verifySeedAvailability(seed) {
-			return seed, nil
+	// Determine all candidate seed cluster matching the shoot's cloud and region.
+	candidates := []*garden.Seed{}
+	for _, seed := range seedList {
+		if seed.DeletionTimestamp == nil && seed.Spec.Cloud.Profile == shoot.Spec.Cloud.Profile && seed.Spec.Cloud.Region == shoot.Spec.Cloud.Region && seed.Spec.Visible != nil && *seed.Spec.Visible && verifySeedAvailability(seed) && verifyNetworkDisjointedNess(seed, shoot) {
+			candidates = append(candidates, seed)
+		}
+	}
+
+	var (
+		bestCandidate *garden.Seed
+		min           = math.MaxInt32
+	)
+
+	// Find the best candidate (i.e. the one managing the smallest number of shoots right now).
+	for _, seed := range candidates {
+		if n := numberOfManagedShoots(shootLister, seed.Name); n < min {
+			bestCandidate = seed
+			min = n
+		}
+	}
+
+	if bestCandidate != nil {
+		return bestCandidate, nil
+	}
+	return nil, errors.New("no adequate seed cluster found for this cloud profile and region")
+}
+
+func numberOfManagedShoots(shootLister gardenlisters.ShootLister, seedName string) int {
```

Comment:
Could you rather fetch the shoots only once and immediately create a `map[string]int` that represents a mapping of seeds to managed shoots? Would also decrease overall computation time + additionally save multiple listing requests (if e.g. sometimes cache is not up to date anymore).